### PR TITLE
Perf: remove getNativeProps usage in mergeProps

### PR DESCRIPTION
**Pull request by dzearing**
This is a first stab at moving closer to the v0 approach of mixing unhandled props into root:

Rather than allow list, which is bulky and long and slow in lookups, mix anything not included in `handledProps`, in addition to things we know aren't valid.

This means that components would need to provide `handledProps` on their config.

### CODE REVIEW

File: packages/react-button/src/components/Button/ButtonBase.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import * as React from 'react';
 import { ButtonProps } from './Button.types';
-import { compose } from '@fluentui/react-compose';
+import { compose, ComposePreparedOptions } from '@fluentui/react-compose';
 import { useButton } from './useButton';
 
 export const ButtonBase = compose<'button', ButtonProps, ButtonProps, {}, {}>(
-  (props, ref, composeOptions) => {
+  (props: ButtonProps, ref: React.Ref<HTMLElement>, composeOptions: ComposePreparedOptions<ButtonProps>) => {
```
Comment (assuncaocharles) [inline packages/react-button/src/components/Button/ButtonBase.tsx:7]: Should it be `HTMLButtonElement`?

```suggestion
  (props: ButtonProps, ref: React.Ref<HTMLButtonElement>, composeOptions: ComposePreparedOptions<ButtonProps>) => {
```

File: packages/react-compose/src/types.ts [hunk 2]
```diff
@@ -99,7 +99,7 @@ export type ComposePreparedOptions<TProps = {}, TState = TProps> = {
   mapPropsToStylesPropsChain: ((props: object) => object)[];
   render: ComposeRenderFunction;
 
-  handledProps: (keyof TProps)[];
+  handledProps: string[];
```
Comment (xugao) [inline packages/react-compose/src/types.ts:102]: why do you change the typings? original looks right to me

File: packages/react-compose/src/types.ts [hunk 3]
```diff
@@ -62,7 +62,7 @@ export type ComposeOptions<TInputProps = {}, TInputStylesProps = {}, TParentProp
 
   mapPropsToStylesProps?: (props: TParentStylesProps & TInputProps) => TInputStylesProps;
 
-  handledProps?: (keyof TInputProps | 'as')[];
+  handledProps?: (keyof (TInputProps & { as: React.ElementType }))[];
```
Comment (xugao) [inline packages/react-compose/src/types.ts:65]:
```suggestion
  handledProps?: (keyof TInputProps | 'as')[];
```

File: packages/react-compose/src/types.ts [hunk 4]
```diff
@@ -99,7 +99,7 @@ export type ComposePreparedOptions<TProps = {}, TState = TProps> = {
   mapPropsToStylesPropsChain: ((props: object) => object)[];
   render: ComposeRenderFunction;
 
-  handledProps: (keyof TProps)[];
+  handledProps: (keyof (TProps & { as: React.ElementType }))[];
```
Comment (xugao) [inline packages/react-compose/src/types.ts:102]:
```suggestion
  handledProps?: (keyof TInputProps | 'as')[];
```

**Pull request merged**